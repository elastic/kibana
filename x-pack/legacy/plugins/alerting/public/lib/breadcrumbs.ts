/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';

interface Breadcrumb {
  text: string;
  href?: string;
}

export const listBreadcrumb: Breadcrumb = {
  text: i18n.translate('xpack.alerting.breadcrumb.listLabel', {
    defaultMessage: 'Alerting',
  }),
  href: '#/management/elasticsearch/alerting/alerts/',
};
