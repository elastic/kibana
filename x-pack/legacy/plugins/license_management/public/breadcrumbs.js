/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';
import { MANAGEMENT_BREADCRUMB } from 'ui/management';
import { BASE_PATH } from '../common/constants';

export function getDashboardBreadcrumbs() {
  return [
    MANAGEMENT_BREADCRUMB,
    {
      text: i18n.translate('xpack.licenseMgmt.dashboard.breadcrumb', {
        defaultMessage: 'License management'
      }),
      href: `#${BASE_PATH}home`
    }
  ];
}

export function getUploadBreadcrumbs() {
  return [
    ...getDashboardBreadcrumbs(),
    {
      text: i18n.translate('xpack.licenseMgmt.upload.breadcrumb', {
        defaultMessage: 'Upload'
      })
    }
  ];
}
