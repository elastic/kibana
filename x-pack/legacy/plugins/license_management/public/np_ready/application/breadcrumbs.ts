/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';

import { BASE_PATH } from '../../../common/constants';

export interface Breadcrumb {
  text: string;
  href: string;
}

export function getDashboardBreadcrumbs(root: Breadcrumb) {
  return [
    root,
    {
      text: i18n.translate('xpack.licenseMgmt.dashboard.breadcrumb', {
        defaultMessage: 'License management',
      }),
      href: `#${BASE_PATH}home`,
    },
  ];
}

export function getUploadBreadcrumbs(root: Breadcrumb) {
  return [
    ...getDashboardBreadcrumbs(root),
    {
      text: i18n.translate('xpack.licenseMgmt.upload.breadcrumb', {
        defaultMessage: 'Upload',
      }),
    },
  ];
}
