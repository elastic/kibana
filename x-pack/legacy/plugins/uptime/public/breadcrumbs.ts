/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';

export interface UMBreadcrumb {
  text: string;
  href?: string;
}

export const overviewBreadcrumb: UMBreadcrumb = {
  text: i18n.translate('xpack.uptime.breadcrumbs.overviewBreadcrumbText', {
    defaultMessage: 'Uptime',
  }),
  href: '#/',
};

export const getOverviewPageBreadcrumbs = (): UMBreadcrumb[] => [overviewBreadcrumb];

export const getMonitorPageBreadcrumb = (name: string): UMBreadcrumb[] => [
  overviewBreadcrumb,
  { text: name },
];
