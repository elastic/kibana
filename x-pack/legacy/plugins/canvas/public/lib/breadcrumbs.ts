/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { ChromeBreadcrumb } from '../../../../../../src/core/public';
import { getCoreStart } from '../legacy';

export const getBaseBreadcrumb = () => ({
  text: 'Canvas',
  href: '#/',
});

export const getWorkpadBreadcrumb = ({
  name = 'Workpad',
  id,
}: { name?: string; id?: string } = {}) => {
  const output: ChromeBreadcrumb = { text: name };
  if (id != null) {
    output.href = `#/workpad/${id}`;
  }
  return output;
};

export const setBreadcrumb = (paths: ChromeBreadcrumb | ChromeBreadcrumb[]) => {
  const setBreadCrumbs = getCoreStart().chrome.setBreadcrumbs;
  setBreadCrumbs(Array.isArray(paths) ? paths : [paths]);
};
