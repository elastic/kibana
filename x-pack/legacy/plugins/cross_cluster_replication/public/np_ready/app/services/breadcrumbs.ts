/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import { i18n } from '@kbn/i18n';
import { ChromeBreadcrumb } from 'src/core/public';

import { ManagementAppMountParams } from '../../../../../../../../src/plugins/management/public';

import { BASE_PATH } from '../../../../common/constants';

let setBreadcrumbs: ManagementAppMountParams['setBreadcrumbs'];

export const setBreadcrumbSetter = ({
  __LEGACY,
}: {
  __LEGACY: {
    chrome: any;
    MANAGEMENT_BREADCRUMB: ChromeBreadcrumb;
  };
}): void => {
  setBreadcrumbs = (crumbs: ChromeBreadcrumb[]) => {
    __LEGACY.chrome.breadcrumbs.set([__LEGACY.MANAGEMENT_BREADCRUMB, ...crumbs]);
  };
};

export const listBreadcrumb = {
  text: i18n.translate('xpack.crossClusterReplication.homeBreadcrumbTitle', {
    defaultMessage: 'Cross-Cluster Replication',
  }),
  href: `#${BASE_PATH}`,
};

export const addBreadcrumb = {
  text: i18n.translate('xpack.crossClusterReplication.addBreadcrumbTitle', {
    defaultMessage: 'Add',
  }),
};

export const editBreadcrumb = {
  text: i18n.translate('xpack.crossClusterReplication.editBreadcrumbTitle', {
    defaultMessage: 'Edit',
  }),
};

export { setBreadcrumbs };
