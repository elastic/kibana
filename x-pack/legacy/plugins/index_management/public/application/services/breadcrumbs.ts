/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import { i18n } from '@kbn/i18n';
import { BASE_PATH } from '../../../common/constants';
import { ChromeStart } from '../../../../../../../src/core/public';

class BreadcrumbService {
  private chrome: ChromeStart | undefined;
  private breadcrumbs: {
    [key: string]: Array<{
      text: string;
      href?: string;
    }>;
  } = {
    management: [],
    home: [],
  };

  public init(chrome: ChromeStart, managementBreadcrumb: any): void {
    this.chrome = chrome;
    this.breadcrumbs.management = [managementBreadcrumb];

    this.breadcrumbs.home = [
      ...this.breadcrumbs.management,
      {
        text: i18n.translate('xpack.idxMgmt.breadcrumb.homeLabel', {
          defaultMessage: 'Index Management',
        }),
        href: `#${BASE_PATH}`,
      },
    ];

    this.breadcrumbs.templates = [
      ...this.breadcrumbs.home,
      {
        text: i18n.translate('xpack.idxMgmt.breadcrumb.templatesLabel', {
          defaultMessage: 'Templates',
        }),
        href: `#${BASE_PATH}templates`,
      },
    ];

    this.breadcrumbs.templateCreate = [
      ...this.breadcrumbs.templates,
      {
        text: i18n.translate('xpack.idxMgmt.breadcrumb.createTemplateLabel', {
          defaultMessage: 'Create template',
        }),
      },
    ];

    this.breadcrumbs.templateEdit = [
      ...this.breadcrumbs.templates,
      {
        text: i18n.translate('xpack.idxMgmt.breadcrumb.editTemplateLabel', {
          defaultMessage: 'Edit template',
        }),
      },
    ];

    this.breadcrumbs.templateClone = [
      ...this.breadcrumbs.templates,
      {
        text: i18n.translate('xpack.idxMgmt.breadcrumb.cloneTemplateLabel', {
          defaultMessage: 'Clone template',
        }),
      },
    ];
  }

  public setBreadcrumbs(type: string): void {
    const newBreadcrumbs = this.breadcrumbs[type]
      ? [...this.breadcrumbs[type]]
      : [...this.breadcrumbs.home];

    // Pop off last breadcrumb
    const lastBreadcrumb = newBreadcrumbs.pop() as {
      text: string;
      href?: string;
    };

    // Put last breadcrumb back without href
    newBreadcrumbs.push({
      ...lastBreadcrumb,
      href: undefined,
    });

    if (this.chrome) {
      this.chrome.setBreadcrumbs(newBreadcrumbs);
    }
  }
}

export const breadcrumbService = new BreadcrumbService();
