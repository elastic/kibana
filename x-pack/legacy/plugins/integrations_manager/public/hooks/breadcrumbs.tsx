/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

// eslint-disable-next-line @kbn/eslint/no-restricted-paths
import { ChromeBreadcrumb } from 'src/core/public/chrome';
import { useCore } from '../hooks';

export function useBreadcrumbs(newBreadcrumbs: ChromeBreadcrumb[]) {
  const { chrome } = useCore();
  return chrome.setBreadcrumbs(newBreadcrumbs);
}
