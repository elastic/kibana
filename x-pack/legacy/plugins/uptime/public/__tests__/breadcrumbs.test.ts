/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { getOverviewPageBreadcrumbs, getMonitorPageBreadcrumb } from '../breadcrumbs';

describe('getOverviewPageBreadcrumbs', () => {
  it('creates the expected breadcrumb with no search or basepath', () => {
    expect(getOverviewPageBreadcrumbs()).toMatchSnapshot();
  });

  it('creates the expected breadcrumb when there is a basepath but no search', () => {
    expect(getOverviewPageBreadcrumbs(undefined, 'xyz')).toMatchSnapshot();
  });

  it('creates the expected breadcrumb with no basepath but has a search', () => {
    expect(
      getOverviewPageBreadcrumbs('?dateRangeStart=now-15m&dateRangeEnd=now')
    ).toMatchSnapshot();
  });

  it('creates the expected breadcrumb with a basepath and a search', () => {
    expect(
      getOverviewPageBreadcrumbs('?dateRangeStart=now-15m&dateRangeEnd=now', 'xyz')
    ).toMatchSnapshot();
  });
});

describe('getMonitorPageBreadcrumb', () => {
  it('creates a monitor page breadcrumb with no search or basepath', () => {
    expect(getMonitorPageBreadcrumb('monitor-id-1234')).toMatchSnapshot();
  });

  it('creates a monitor page breadcrumb with a search but no basepath', () => {
    expect(
      getMonitorPageBreadcrumb('monitor-id-1234', '?dateRangeEnd=now&dateRangeStart=now-15m')
    ).toMatchSnapshot();
  });

  it('creates a monitor page breadcrumb with a basepath but no search', () => {
    expect(getMonitorPageBreadcrumb('monitor-id-1234', undefined, 'xyz')).toMatchSnapshot();
  });

  it('creates a monitor page breadcrumb with a basepath a search', () => {
    expect(
      getMonitorPageBreadcrumb('monitor-id-1234', '?dateRangeStart=now-15m&dateRangeEnd=now', 'xyz')
    ).toMatchSnapshot();
  });
});
