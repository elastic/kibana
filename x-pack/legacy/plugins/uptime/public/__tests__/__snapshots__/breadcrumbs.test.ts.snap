// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMonitorPageBreadcrumb creates a monitor page breadcrumb with a basepath a search 1`] = `
Array [
  Object {
    "href": "/xyz/app/uptime/?dateRangeStart=now-15m&dateRangeEnd=now",
    "text": "Uptime",
  },
  Object {
    "text": "monitor-id-1234",
  },
]
`;

exports[`getMonitorPageBreadcrumb creates a monitor page breadcrumb with a basepath but no search 1`] = `
Array [
  Object {
    "href": "/xyz/app/uptime/",
    "text": "Uptime",
  },
  Object {
    "text": "monitor-id-1234",
  },
]
`;

exports[`getMonitorPageBreadcrumb creates a monitor page breadcrumb with a search but no basepath 1`] = `
Array [
  Object {
    "href": "/app/uptime/?dateRangeEnd=now&dateRangeStart=now-15m",
    "text": "Uptime",
  },
  Object {
    "text": "monitor-id-1234",
  },
]
`;

exports[`getMonitorPageBreadcrumb creates a monitor page breadcrumb with no search or basepath 1`] = `
Array [
  Object {
    "href": "/app/uptime/",
    "text": "Uptime",
  },
  Object {
    "text": "monitor-id-1234",
  },
]
`;

exports[`getOverviewPageBreadcrumbs creates the expected breadcrumb when there is a basepath but no search 1`] = `
Array [
  Object {
    "href": "/xyz/app/uptime/",
    "text": "Uptime",
  },
]
`;

exports[`getOverviewPageBreadcrumbs creates the expected breadcrumb with a basepath and a search 1`] = `
Array [
  Object {
    "href": "/xyz/app/uptime/?dateRangeStart=now-15m&dateRangeEnd=now",
    "text": "Uptime",
  },
]
`;

exports[`getOverviewPageBreadcrumbs creates the expected breadcrumb with no basepath but has a search 1`] = `
Array [
  Object {
    "href": "/app/uptime/?dateRangeStart=now-15m&dateRangeEnd=now",
    "text": "Uptime",
  },
]
`;

exports[`getOverviewPageBreadcrumbs creates the expected breadcrumb with no search or basepath 1`] = `
Array [
  Object {
    "href": "/app/uptime/",
    "text": "Uptime",
  },
]
`;
